backends/cuda/ceed-cuda.c

```diff
@@ -110,8 +121,17 @@ int CeedGetKernelCuda(Ceed ceed, CUmodule module, const char *name,
 //------------------------------------------------------------------------------
 int CeedRunKernelCuda(Ceed ceed, CUfunction kernel, const int gridSize,
                       const int blockSize, void **args) {
-  CeedChk_Cu(ceed, cuLaunchKernel(kernel, gridSize, 1, 1, blockSize, 1,
-                                  1, 0, NULL, args, NULL));
+  CUresult result = cuLaunchKernel(kernel, gridSize, 1, 1, blockSize, 1,
+                                   1, 0, NULL, args, NULL);
+  if (result == CUDA_ERROR_LAUNCH_OUT_OF_RESOURCES) {
+    int max_threads_per_block, shared_size_bytes, num_regs;
+    cuFuncGetAttribute(&max_threads_per_block, CU_FUNC_ATTRIBUTE_MAX_THREADS_PER_BLOCK, kernel);
```

Comment:
Seems we should just set the number of threads per block according to this, rather than the max system reported.

Reply:
Quite often the number of threads is specific to the number of quads, but this would be a good default, for instance to a function where we don't provide the blocksize.